src/software/simulation/er_force_simulator.cpp

```diff
@@ -0,0 +1,296 @@
+    parser.ParseFromString(s, &er_force_sim_setup);
+
+    er_force_sim =
+        std::make_unique<camun::simulator::Simulator>(er_force_sim_setup, true);
+
+    auto simulator_setup_command = std::make_shared<amun::Command>();
+    simulator_setup_command->mutable_simulator()->set_enable(true);
+    // start with default robots, take ER-Force specs.
+    robot::Specs ERForce;
+    robotSetDefault(&ERForce);
+
+    Team friendly_team = Team();
+    Team enemy_team    = Team();
+    Ball ball          = Ball(Point(), Vector(), Timestamp::fromSeconds(0));
+
+    World world = World(field, ball, friendly_team, enemy_team);
+
+    // TODO (#2283): remove this initialization when addYellowRobots and addBlueRobots are
+    // implemented
+    auto* team_blue   = simulator_setup_command->mutable_set_team_blue();
+    auto* team_yellow = simulator_setup_command->mutable_set_team_yellow();
+    for (auto* team : {team_blue, team_yellow})
+    {
+        for (int i = 0; i < 11; ++i)
+        {
+            auto* robot = team->add_robot();
+            robot->CopyFrom(ERForce);
+            robot->set_id(i);
+        }
+    }
+    er_force_sim->handleSimulatorSetupCommand(simulator_setup_command);
+    er_force_sim->seedPRGN(17);
+
+    // TODO (#2283): remove this initialization when addYellowRobots and addBlueRobots are
+    // implemented
+    for (unsigned int i = 0; i < 11; i++)
+    {
+        auto blue_simulator_robot = std::make_shared<ErForceSimulatorRobot>(
+            RobotStateWithId{.id          = i,
+                             .robot_state = RobotState(Point(), Vector(), Angle::zero(),
+                                                       AngularVelocity::zero())},
+            robot_constants, wheel_constants);
+        auto yellow_simulator_robot = std::make_shared<ErForceSimulatorRobot>(
+            RobotStateWithId{.id          = i,
+                             .robot_state = RobotState(Point(), Vector(), Angle::zero(),
+                                                       AngularVelocity::zero())},
+            robot_constants, wheel_constants);
+
+        auto blue_firmware_robot = ErForceSimulatorRobotSingleton::createFirmwareRobot();
+        auto blue_firmware_ball  = SimulatorBallSingleton::createFirmwareBall();
+        auto yellow_firmware_robot =
+            ErForceSimulatorRobotSingleton::createFirmwareRobot();
+        auto yellow_firmware_ball = SimulatorBallSingleton::createFirmwareBall();
+
+        FirmwareWorld_t* blue_firmware_world_raw = app_firmware_world_create(
+            blue_firmware_robot.release(), blue_firmware_ball.release(),
+            &(ErForceSimulator::getCurrentFirmwareTimeSeconds));
+        auto blue_firmware_world = std::shared_ptr<FirmwareWorld_t>(
+            blue_firmware_world_raw, FirmwareWorldDeleter());
+
+        FirmwareWorld_t* yellow_firmware_world_raw = app_firmware_world_create(
+            yellow_firmware_robot.release(), blue_firmware_ball.release(),
```

Comment:
Should `blue_firmware_ball.release()` be `yellow_firmware_ball.release()` here?

Reply:
nice catch fixed